index.bs

```diff
@@ -3484,6 +3641,301 @@ partial dictionary MLOpSupportLimits {
+    **Arguments:**
+        - <dfn>input</dfn>: an {{MLOperand}}. The input tensor.
+        - <dfn>scale</dfn>: an {{MLOperand}}. The scale tensor to multiply each input value by after adjusting by the zero point. It must be [=blockwise broadcastable=] with the input.
+        - <dfn>zeroPoint</dfn>: an {{MLOperand}}. The zero point tensor to subtract from each input value. It has the same [=MLOperand/shape=] as the scale.
+        - <dfn>options</dfn>: an {{MLOperatorOptions}}. Specifies the optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The output tensor that contains the dequantized values.
+</div>
+
+<table id=constraints-dequantizelinear class='data' link-for="MLGraphBuilder/dequantizeLinear(input, scale, zeroPoint, options)">
+  <caption>Constraints for {{MLGraphBuilder/dequantizeLinear()}}</caption>
+  <thead>
+    <tr>
+      <th>operand</th>
+      <th>[=/allowed data types=]</th>
+      <th>[=/allowed ranks=]</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>{{input}}</td>
+    <td>{{MLOperandDataType/"uint8"}}, {{MLOperandDataType/"uint32"}}, {{MLOperandDataType/"int32"}}</td>
+    <td>[=/any rank|N=]</td>
+  </tr>
+  <tr>
+    <td>{{scale}}</td>
+    <td>{{MLOperandDataType/"float32"}}, {{MLOperandDataType/"float16"}}</td>
+    <td>[=/same rank as|same as=] {{input}}</td>
+  </tr>
+  <tr>
+    <td>{{zeroPoint}}</td>
+    <td>[=/same type as|same as=] {{input}}</td>
+    <td>[=/same rank as|same as=] {{input}}</td>
+  </tr>
+  <tr>
+    <td>*output*</td>
+    <td>[=/same type as|same as=] {{scale}}</td>
+    <td>[=/same rank as|same as=] {{input}}</td>
+  </tr>
+</table>
+
+{{MLQuantizationSupportLimits}} has the following members:
+<dl dfn-type=dict-member dfn-for=MLQuantizationSupportLimits>
+    : <dfn>input</dfn>
+    :: {{MLSupportLimits}} for input operand.
+    : <dfn>scale</dfn>
+    :: {{MLSupportLimits}} for scale operand.
+    : <dfn>zeroPoint</dfn>
+    :: {{MLSupportLimits}} for zeroPoint operand.
+    : <dfn>output</dfn>
+    :: {{MLSupportLimits}} for output operand.
+</dl>
+
+{{MLOpSupportLimits}} has the following member for {{MLGraphBuilder/dequantizeLinear()}}:
+<dl dfn-type=dict-member dfn-for=MLOpSupportLimits>
+    : <dfn>dequantizeLinear</dfn>
+    :: Support limits for operator {{MLGraphBuilder/dequantizeLinear()}}.
+</dl>
+
+<details open algorithm>
+  <summary>
+    The <dfn method for=MLGraphBuilder>dequantizeLinear(|input|, |scale|, |zeroPoint|, |options|)</dfn> method steps are:
+  </summary>
+    1. If [=this=].{{MLGraphBuilder/[[hasBuilt]]}} is true, then [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
+    1. If [=MLGraphBuilder/validating operand=] with [=this=] and any of |input|, |scale|, and |zeroPoint| returns false, then [=exception/throw=] a {{TypeError}}.
+    1. If |scale|'s [=MLOperand/rank=] or |zeroPoint|'s [=MLOperand/rank=] mismatches |input|'s [=MLOperand/rank=], then [=exception/throw=] a {{TypeError}}.
```

Comment:
"mismatches" is not used elsewhere in the spec. "is not equal to" is more common.

For rank specifically, you could also use _"If scale's rank is not its allowed rank (according to this table), then throw a TypeError."_

(I see 4 total uses of "mismatches" introduced by this PR)